lib/components/SignInWithGoogle/SignInWithGoogle.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState, useEffect } from 'react'
+import Button from '../Button'
+import Loader from '../Loader'
+import Utils from 'utils'
+
+import './signInWithGoogle.scss'
+
+interface TokenResponse {
+  access_token: string
+  expires_in: number
+  hd: string
+  prompt: string
+  token_type: string
+  scope: string
+  state?: string
+  error?: string
+  error_description?: string
+  error_uri?: string
+}
+
+interface TokenClient {
+  requestAccessToken(): void
+}
+
+interface SignInWithGoogleProps {
+  googleApi: string
+  clientId: string
+  scope: string
+  apiCall: (data: { provider: string; token: string }) => Promise<unknown>
+  setUserFunc?: (data: unknown) => void
+}
+
+function SignInWithGoogle({
+  googleApi,
+  clientId,
+  scope,
+  apiCall,
+  setUserFunc,
+  ...rest
+}: SignInWithGoogleProps) {
+  const [client, setClient] = useState<TokenClient | null>(null)
+  const [googleLoaded, setGoogleLoaded] = useState(false)
+
+  const handleGoogleSignIn = async (tokenResponse: TokenResponse) => {
+    if (tokenResponse && tokenResponse.access_token) {
```

Comment:
Lets handle unhappy flow first, so we have less indentation
```suggestion
    if (!tokenResponse?.access_token) {
      return
    }
    ... try catch code
```